torch/lib/THD/base/data_channels/DataChannelNccl.cpp

```diff
@@ -577,7 +576,7 @@ void DataChannelNccl::broadcast(std::vector<at::Tensor>& data,
   auto events = ncclResourcePair.second;
 
   // Guard GPU device
-  AutoGPU gpuGuard;
+  at::DeviceGuard gpuGuard;
```

Comment:
@goldsborough this does not compile. See line 587

Reply:
fixing